### PR TITLE
CIV-6434 assign claim to defendant

### PR DESCRIPTION
**Pull request by TatsyMaslova**
### JIRA link ###

https://tools.hmcts.net/jira/browse/CIV-6434

### Change description ###
- make pin and post journey public (first contact pages)
- redirect after idam login to assign claim to defendant and then redirect to dashboard
- amend and add unit tests

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ x] No
```

### CODE REVIEW

File: src/main/routes/features/claimAssignment/assignClaimController.ts [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+import {AppRequest} from 'models/AppRequest';
+
+const assignClaimController = Router();
+const civilServiceApiBaseUrl = config.get<string>('services.civilService.url');
+const civilServiceClient: CivilServiceClient = new CivilServiceClient(civilServiceApiBaseUrl);
+
+assignClaimController.get(ASSIGN_CLAIM_URL, async ( req:AppRequest, res) => {
+  const claimId = <string>req.query?.id;
+  try{
+    if(claimId){
```
Comment (PaulH-HMCTS) [inline src/main/routes/features/claimAssignment/assignClaimController.ts:14]:
```suggestion
    if (claimId) {
```

File: src/main/routes/features/claimAssignment/assignClaimController.ts [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+      await civilServiceClient.assignDefendantToClaim(claimId, req);
+    }
+  }finally {
```
Comment (PaulH-HMCTS) [inline src/main/routes/features/claimAssignment/assignClaimController.ts:17]:
```suggestion
  } finally {
```

File: src/test/unit/routes/features/claimAssignment/claimAssignmentController.test.ts [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+      nock(civilServiceUrl)
+        .post('/assignment/case/123/RESPONDENTSOLICITORONE')
+        .reply(200);
+      request(app).post(ASSIGN_CLAIM_URL+'?123')
+        .expect((res) => {
+          expect(res.status).toBe(200);
+          expect(res.get('location')).toBe(DASHBOARD_URL);
+        });
+    });
+    it('sbould not call civil service api and redirect to dashboard', async () =>{
```
Comment (PaulH-HMCTS) [inline src/test/unit/routes/features/claimAssignment/claimAssignmentController.test.ts:31]:
```suggestion
    it('should not call civil service api and redirect to dashboard', async () =>{
```

File: src/main/app/client/civilServiceClient.ts [hunk 4]
```diff
@@ -223,4 +224,14 @@ export class CivilServiceClient {
       throw error;
     }
   }
+
+  async assignDefendantToClaim(claimId:string, req:AppRequest): Promise<void> {
+    try{
+      await this.client.post(ASSIGN_CLAIM_TO_DEFENDANT.replace(':caseId', claimId),{},
```
[the remark this answers is not in the document]
Comment (TatsyMaslova) [reply src/main/app/client/civilServiceClient.ts]: done

**Review by laperla11 — DISMISSED** [review]

File: src/main/routes/features/claimAssignment/assignClaimController.ts [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+const assignClaimController = Router();
+const civilServiceApiBaseUrl = config.get<string>('services.civilService.url');
+const civilServiceClient: CivilServiceClient = new CivilServiceClient(civilServiceApiBaseUrl);
+
+assignClaimController.get(ASSIGN_CLAIM_URL, async ( req:AppRequest, res) => {
+  const claimId = <string>req.query?.id;
+  try{
+    if (claimId) {
+      await civilServiceClient.assignDefendantToClaim(claimId, req);
+    }
```
Comment (LeonardoPalmeiro) [inline src/main/routes/features/claimAssignment/assignClaimController.ts:16]: I don't know if in here could produce an exception if yes, please add catch statement.

**Comment by LeonardoPalmeiro** [comment]
I saw in Sonar there are some methods without coverage, please check sonar result.

File: src/main/routes/features/claimAssignment/assignClaimController.ts (hunk 5 above)
Comment (TatsyMaslova) [reply src/main/routes/features/claimAssignment/assignClaimController.ts:16]: in both cases whether there is an error or it is successful we are going to dashboard. the error is already logged in the client when it occurs hence we do not need a catch

**Comment by TatsyMaslova** [comment]
@LeonardoPalmeiro yes i saw Sonar. and tried to cover everything that is possible to cover. However some logic in oidc is very difficult to cover without changing it like Maro did. I am reluctant to do so. Perhaps will ask @PaulH-HMCTS advice

**Review by laperla11 — APPROVED** [review]

**Review by portizdev — DISMISSED** [review]

**Review by LeonardoPalmeiro — APPROVED** [review]
LGTM

**Review by portizdev — DISMISSED** [review]

**Review by PaulH-HMCTS — APPROVED** [review]

**Review by portizdev — APPROVED** [review]

**Review by LeonardoPalmeiro — APPROVED** [review]

**Pull request merged**